### PR TITLE
🚀 Implementing Nameable and Decorators 🚀

### PR DESCRIPTION
**Pull request by jicamargo**
# 🚀 Implementing Nameable and Decorators 🚀

This PR introduces the `Nameable` interface and decorators to enhance the functionality of the `Person` class. It adds two decorators: `CapitalizeDecorator` and `TrimmerDecorator`. The decorators provide additional features to the `correct_name` method of the `Person` class.

Changes:
1. Created the `Nameable` class and implemented the `correct_name` method with `NotImplementedError` to raise an exception.
2. Modified the `Person` class to inherit from `Nameable` and implemented the `correct_name` method to return the name attribute.
3. Created the base `Decorator` class, which inherits from `Nameable`. It receives a `nameable` object in the constructor.
4. Implemented the `correct_name` method in the `Decorator` class to return the result of the `correct_name` method of the `@nameable`.
5. Created the `CapitalizeDecorator` class, which inherits from `Decorator`. Implemented the `correct_name` method to capitalize the output of `@nameable.correct_name`.
6. Created the `TrimmerDecorator` class, which inherits from `Decorator`. Implemented the `correct_name` method to ensure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it exceeds 10 characters, it is trimmed.
7. check_decorator.br is a program to test the decorators

Please review the changes and let me know if there's anything else to be added. 
Thank you! 🙌😊

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# person.rb
+
+#
+# Interface Class Nameable, for decorator design
+#
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline person.rb:10]: - [ ] [OPTIONAL] It would be nice if you put this in a separate file to make your code base readable

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  def initialize(person)
+    super()
+    @person = person
+  end
+
+  # The Decorator delegates all work to the wrapped component (person)
+  def correct_name
+    @person.correct_name
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline person.rb:56]: - [ ] [OPTIONAL] It would be nice if you put this in a separate file to make your code base readable

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+# Concrete Decorators
+# call the wrapped object and alter its result in some way.
+#
+# Decorator class: CapitalizeDecorator
+#
+class CapitalizeDecorator < PersonDecorator
+  def correct_name
+    @person.correct_name.capitalize
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline person.rb:67]: - [ ] [OPTIONAL] It would be nice if you put this in a separate file to make your code base readable

File: person.rb [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+#
+# Decorator class: TrimmerDecorator
+# makes sure that the output of correct_name has a maximum of 10 characters.
+#
+class TrimmerDecorator < PersonDecorator
+  def correct_name
+    name = @person.correct_name
+    name.length > 10 ? name[0..9] : name
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline person.rb:80]: - [ ] [OPTIONAL] It would be nice if you put this in a separate file to make your code base readable

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @jicamargo , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 4 above)
Comment (jicamargo) [reply person.rb:80]: thank you. I'll take it into account

**Pull request closed without merging**